addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -650,6 +650,16 @@ export class SelfOrder extends Reactive {
         return result;
     }
 
+    verifyPriceLoading() {
+        if (this.priceLoading) {
```

Comment:
the `pay` button should be disabled in this case. ( `t-att-disabled="this.priceLoading` )
Allowing the user to click `pay` only to show a notification is weird

Reply:
@vlst-odoo Thanks for your feedback.
The button is dynamic and is inside the order widget, so it's not possible to add this condition directly to the button without any context. We need to apply it in specific cases.
This issue occurs when the user clicks on the pay button very quickly. Showing a notification and preventing the process from going forward is a mechanism to ensure the payment step is not missed, without causing additional issues.

Reply:
why would this not work ?
```xml
            <button t-att-disabled="selfOrder.priceLoading" t-attf-class="cart btn btn-primary btn-lg flex-grow-1 flex-md-grow-0 {{ buttonToShow.disabled ? 'disabled' : '' }}" t-on-click="props.action" t-esc="buttonToShow.label"/>
```

Reply:
@vlst-odoo Then it disables the "Order" button when the price is not loaded, and it would be unnecessary. We only need to prevent bypassing the payment step when the price is not loaded. Adding this condition to the button would disable it in all cases when the price is not loaded, which is not required.